backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java

```diff
@@ -432,20 +553,28 @@ private static AutogenGitCodeDiff getGitDiff(
         .setStatus(AutogenDiffStatusEnumDiffStatus.fromProto(DiffStatus.CONFLICTED));
   }
 
-  private static AutogenDatasetDiff getDatasetConflictBlob(
+  private static List<AutogenDatasetDiff> getDatasetConflictBlob(
       AutogenDatasetDiff a, AutogenDatasetDiff b, AutogenDatasetBlob c) {
-    return Utils.removeEmpty(
-        new AutogenDatasetDiff()
-            .setPath(
-                getPathDatasetConflictBlob(
-                    a == null ? null : a.getPath(),
-                    b == null ? null : b.getPath(),
-                    c == null ? null : c.getPath()))
-            .setS3(
-                getS3DatasetConflictBlob(
-                    a == null ? null : a.getS3(),
-                    b == null ? null : b.getS3(),
-                    c == null ? null : c.getS3())));
+    AutogenPathDatasetDiff pathConflict =
+        getPathDatasetConflictBlob(
+            a == null ? null : a.getPath(),
+            b == null ? null : b.getPath(),
+            c == null ? null : c.getPath());
+
+    AutogenS3DatasetDiff s3Conflict =
+        getS3DatasetConflictBlob(
+            a == null ? null : a.getS3(),
+            b == null ? null : b.getS3(),
+            c == null ? null : c.getS3());
+    if (pathConflict != null && s3Conflict != null) {
+
+      return Arrays.asList(
+          Utils.removeEmpty(new AutogenDatasetDiff().setPath(pathConflict)),
+          Utils.removeEmpty(new AutogenDatasetDiff().setS3(s3Conflict)));
+    } else {
+      return Collections.singletonList(
+          Utils.removeEmpty(new AutogenDatasetDiff().setPath(pathConflict).setS3(s3Conflict)));
```

Comment:
If `pathConflict` & `s3Conflict` i think this line should throw an error because proto does not accept the null value. just need to cross-check.

Reply:
that was the advantage of the Autogen classes.